restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java

```diff
@@ -0,0 +1,752 @@
+package io.stargate.web.docsapi.resources;
+
+import com.datastax.oss.driver.api.core.NoNodeAvailableException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.stargate.auth.UnauthorizedException;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.exception.DocumentAPIRequestException;
+import io.stargate.web.docsapi.service.DocumentService;
+import io.stargate.web.docsapi.service.filter.FilterCondition;
+import io.stargate.web.models.ResponseWrapper;
+import io.stargate.web.resources.Db;
+import io.swagger.annotations.*;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.ws.rs.*;
+import javax.ws.rs.core.*;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/v2/namespaces")
+@Api(
+    produces = MediaType.APPLICATION_JSON,
+    consumes = MediaType.APPLICATION_JSON,
+    tags = {"documents"})
+@Produces(MediaType.APPLICATION_JSON)
+public class DocumentResourceV2 {
+  @Inject private Db dbFactory;
+  private static final Logger logger = LoggerFactory.getLogger(DocumentResourceV2.class);
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private final DocumentService documentService = new DocumentService();
+  private final int DEFAULT_PAGE_SIZE = 100;
+
+  private JsonNode wrapResponse(JsonNode node, String id) {
+    ObjectNode wrapperNode = mapper.createObjectNode();
+
+    if (id != null) {
+      wrapperNode.set("documentId", TextNode.valueOf(id));
+    }
+    if (node != null) {
+      wrapperNode.set("data", node);
+    }
+    return wrapperNode;
+  }
+
+  private JsonNode wrapResponse(JsonNode node, String id, String pagingState) {
+    ObjectNode wrapperNode = mapper.createObjectNode();
+
+    if (id != null) {
+      wrapperNode.set("documentId", TextNode.valueOf(id));
+    }
+    if (node != null) {
+      wrapperNode.set("data", node);
+    }
+    wrapperNode.set("pageState", TextNode.valueOf(pagingState));
+    return wrapperNode;
+  }
+
+  @POST
+  @ApiOperation(
+      value = "Create a new document",
+      notes = "Auto-generates an ID for the newly created document")
```

Comment:
What about the response?

Reply:
Since I'm returning a JsonNode, I can't just inline `JsonNode.class` as the example response - should I make a separate package with response examples that are just beans?

Reply:
There will be a follow up PR as discussed.